packages/react-checkbox/src/components/Checkbox/Checkbox.types.ts

```diff
@@ -47,12 +44,17 @@ export interface CheckboxOnChangeData {
 
 export type CheckboxSlots = {
   /**
-   * The root element of the checkbox is its `<label>`.
+   * The root element of the Checkbox.
    *
    * The root slot receives the `className` and `style` specified directly on the `<Checkbox>`.
    * All other native props will be applied to the primary slot: `input`
    */
-  root: ObjectShorthandProps<LabelProps> | IntrinsicShorthandProps<'span'>;
+  root: IntrinsicShorthandProps<'span'>;
+
+  /**
+   * The Checkbox's label.
+   */
+  label?: ObjectShorthandProps<LabelProps>;
 
   /**
    * Hidden input that handles the checkbox's functionality.
```

Comment:
Consider if the indicator slot should be called checkmark as that is how the comment describes it.

Reply:
Technically the indicator slot is the box, and its child is the checkmark, which is why it wasn't just called checkmark. That said, because of the shorthand syntax, `checkmark={<MyCheckmarkIcon />}` seems more ergonomic.

I logged #20994 to follow up.